### PR TITLE
[Android][Feature][56588] feature/create_account_screen

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Task 56588: Account Screen](https://edu-redmine.sun-asterisk.vn/issues/56588)
- [#Task 56589: Changed Password Screen](https://edu-redmine.sun-asterisk.vn/issues/56589)
## WHAT
- Display data of user
- feature changed password
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by nguyennhatminh230801 — APPROVED** [review]

**Review by ndt0409 — APPROVED** [review]

File: app/src/main/java/com/atom/android/lebo/data/repository/login/ILoginRepository.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.atom.android.lebo.base.BaseResponse
+import com.atom.android.lebo.model.LoginEntity
+import io.reactivex.rxjava3.core.Single
+
+interface ILoginRepository {
+
+    fun login(loginEntity: LoginEntity): Single<BaseResponse<String?>>
+    fun requestOTPLogin(email: String?): Single<BaseResponse<String?>>
+    fun checkLogin(): Single<BaseResponse<String?>>
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/data/repository/login/ILoginRepository.kt:11]: những action ko cần authorize thì tạo riêng 1 interface api service riêng nhé 
dạng như 
interface ApiService {
      interface AuthApiService{}
      interface UnAuthApiService {}
}

File: app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.atom.android.lebo.ui.main.MainViewModel
+import com.atom.android.lebo.utils.extensions.loadImage
+import com.atom.android.lebo.utils.extensions.openDialogQuestion
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class AccountFragment : BaseFragment<FragmentAccountBinding>(FragmentAccountBinding::inflate) {
+
+    override val viewModel by viewModel<AccountViewModel>()
+
+    private val activityViewModel by viewModel<MainViewModel>()
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt]:
```suggestion
    private val activityViewModel by activityViewmodel<MainViewModel>()
```

File: app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        viewModel.logout()
+                    }
+                }
+            }
+
+            btnLogin.setOnClickListener {
+                findNavController().navigate(R.id.action_navigation_account_to_navigation_login)
+            }
+
+            swipeRefreshLayout.setOnRefreshListener {
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt:67]: ko gán lại isRefresh = false à

File: app/src/main/java/com/atom/android/lebo/ui/authentication/login/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        viewModel.validateForm(context, emailObservable, passwordObservable)
+
+        binding.btnLogin.setOnClickListener {
+            val email = binding.textInputLayoutEmail.editText?.text.toString()
+            val password = binding.textInputLayoutPassword.editText?.text.toString()
+            viewModel.login(email, password)
+        }
+        binding.textViewForgotPassword.setOnClickListener {
+            val email = binding.textInputLayoutEmail.editText?.text.toString()
+            val validateEmailResult = email.isValidEmail(context)
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/authentication/login/LoginFragment.kt:82]: dùng mediatorlivedata để validate thử đi em

File: app/src/main/java/com/atom/android/lebo/ui/changepassword/ChangedPasswordViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                confirmPassword
+            ) { _oldPassword, _newPassword, _confirmPassword ->
+                val validOldPasswordResult = _oldPassword.toString().isValidPassword(context)
+                val validNewPasswordResult = _newPassword.toString().isValidPassword(context)
+                val validConfirmPasswordResult =
+                    _confirmPassword.toString()
+                        .isValidConfirmPassword(context, _newPassword.toString())
+                _oldPasswordValid.value = validOldPasswordResult
+                _newPasswordValid.value = validNewPasswordResult
+                _confirmPasswordValid.value = validConfirmPasswordResult
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/lebo/ui/changepassword/ChangedPasswordViewModel.kt:70]: đây nguyên đống này , dùng MediatorLiveData để map lại

File: app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt (hunk 3 above)
Comment (hungvv-1301) [reply app/src/main/java/com/atom/android/lebo/ui/account/AccountFragment.kt:67]: gán dòng 43 rùi mà a

**Pull request merged**